webplugin/js/app/kommunicate.js

```diff
@@ -826,4 +828,23 @@ $applozic.extend(true, Kommunicate, {
         }
         return siblings;
     },
+    btnClickPreChatOpen: function () {
+        var kommunicateIframe = parent.document.getElementById(
+            'kommunicate-widget-iframe'
+        );
+        var modal = document.getElementById('km-chat-login-modal');
+        var closeBtn = document.getElementById('km-popup-close-button');
+        var popupMsg = document.getElementById('chat-popup-widget-container');
+        kommunicateIframe.className =
+            'kommunicate-custom-iframe km-iframe-dimension-with-popup kommunicate-iframe-enable-media-query';
+        if (modal) {
+            modal.style.display = 'block';
+        }
+        if (closeBtn) {
+            closeBtn.style.display = 'flex';
+        }
+        if (popupMsg) {
+            popupMsg.style.display = 'none';
+        }
```

Comment:
I guess we have to just open the widget right?
can't that be done by triggering the click event of the launcher?